src/components/ReportActionItem/MoneyRequestView.js

```diff
@@ -65,13 +78,19 @@ function MoneyRequestView({report, parentReport, shouldShowHorizontalRule, trans
         currency: transactionCurrency,
         comment: transactionDescription,
         merchant: transactionMerchant,
+        category: transactionCategory,
     } = ReportUtils.getTransactionDetails(transaction);
     const isEmptyMerchant =
         transactionMerchant === '' || transactionMerchant === CONST.TRANSACTION.UNKNOWN_MERCHANT || transactionMerchant === CONST.TRANSACTION.PARTIAL_TRANSACTION_MERCHANT;
     const formattedTransactionAmount = transactionAmount && transactionCurrency && CurrencyUtils.convertToDisplayString(transactionAmount, transactionCurrency);
 
     const isSettled = ReportUtils.isSettled(moneyRequestReport.reportID);
     const canEdit = ReportUtils.canEditMoneyRequest(parentReportAction);
+    // A flag for verifying that the current report is a sub-report of a workspace chat
+    const isPolicyExpenseChat = useMemo(() => ReportUtils.isPolicyExpenseChat(ReportUtils.getRootParentReport(report)), [report]);
```

Comment:
I would log if the `useMemo` is needed here, because `report` is an obj that changes frequently also on every rerender, and `isPolicyExpenseChat` isn't an expensive function

Reply:
So, I need `useMemo` here, because `getRootParentReport` is a kind of expensive functions. It uses a recursion to get the most root report. Also, the `report` object is not being updated frequently of this component, I tested. So, that's why.
 
<img width="628" alt="Screenshot 2023-09-19 at 11 40 27" src="https://github.com/Expensify/App/assets/57314631/1b0637ef-2203-4c69-b424-7322e859900f">

Reply:
Awesome